adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -128,7 +129,32 @@ AuthenticationResult acquireTokenWithRefreshToken(final String refreshToken)
                 Logger.i(TAG, "Refresh token is not returned or empty", "");
                 result.setRefreshToken(refreshToken);
             }
-        } catch (final IOException | AuthenticationException exc) {
+        } catch (final AuthenticationException exc) {
+            //if get 503,504,500 && outageMode is on, return the stale token
+            if (exc.getCode().equals(ADALError.SERVER_NOT_RESPONDING) && mOutageModeIsOn) {
+                final TokenCacheItem accessTokenItem = mTokenCacheAccessor.getRegularRefreshTokenCacheItem(mAuthRequest.getResource(),
```

Comment:
@addev-ashish and I talked about this, to make the code more readable, we should wrap the logic here in a separate method instead of having a big block of code in the catch itself.